server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/qualitygate/QualityGateHolderImpl.java

```diff
@@ -19,44 +19,46 @@
  */
 package org.sonar.server.computation.task.projectanalysis.qualitygate;
 
-import com.google.common.base.Optional;
-import javax.annotation.CheckForNull;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.server.qualitygate.EvaluatedQualityGate;
 
-import static com.google.common.base.Optional.absent;
-import static com.google.common.base.Optional.of;
 import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 public class QualityGateHolderImpl implements MutableQualityGateHolder {
-  private boolean initialized = false;
-  @CheckForNull
+  @Nullable
   private Optional<QualityGate> qualityGate;
+  @Nullable
+  private Optional<EvaluatedQualityGate> evaluation;
 
   @Override
-  public void setQualityGate(QualityGate qualityGate) {
+  public void setQualityGate(QualityGate g) {
     // fail fast
-    requireNonNull(qualityGate);
-    checkNotInitialized();
+    requireNonNull(g);
+    checkState(qualityGate == null, "QualityGateHolder can be initialized only once");
 
-    this.initialized = true;
-    this.qualityGate = of(qualityGate);
+    this.qualityGate = Optional.of(g);
   }
 
   @Override
-  public void setNoQualityGate() {
-    checkNotInitialized();
-
-    this.initialized = true;
-    this.qualityGate = absent();
+  public Optional<QualityGate> getQualityGate() {
+    checkState(qualityGate != null, "QualityGate has not been set yet");
+    return qualityGate;
   }
 
-  private void checkNotInitialized() {
-    checkState(!initialized, "QualityGateHolder can be initialized only once");
+  @Override
+  public void setEvaluation(EvaluatedQualityGate g) {
+    // fail fast
+    requireNonNull(g);
+    checkState(evaluation == null, "QualityGateHolder evaluation can be initialized only once");
+
+    this.evaluation = Optional.of(g);
   }
 
   @Override
-  public Optional<QualityGate> getQualityGate() {
-    checkState(initialized, "QualityGate has not been set yet");
-    return qualityGate;
+  public Optional<EvaluatedQualityGate> getEvaluation() {
+    checkState(evaluation != null, "Evaluation of QualityGate has not been set yet");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this null-check of an "Optional". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2789)